pulsar-flink/src/main/java/org/apache/flink/streaming/connectors/pulsar/PulsarSourceBuilder.java

```diff
@@ -277,9 +278,27 @@ private PulsarSourceBuilder(DeserializationSchema<T> deserializationSchema) {
                 "At least one topic or topics pattern is required");
         Preconditions.checkArgument(StringUtils.isNotBlank(this.consumerConfigurationData.getSubscriptionName()),
                 "a subscription name is required");
+
+        setTransientFields();
+
         return new PulsarConsumerSource<>(this);
     }
 
+    private void setTransientFields() throws PulsarClientException {
+        setAuth();
+    }
+
+    private void setAuth() throws PulsarClientException{
+        if (StringUtils.isBlank(this.clientConfigurationData.getAuthPluginClassName())
+                && this.clientConfigurationData.getAuthParams() == null || this.clientConfigurationData.getAuthParams().isEmpty())
```

Comment:
```suggestion
                && StringUtils.isBlank(this.clientConfigurationData.getAuthParams()))
```

Reply:
@merlimat this returns a map so thats not possible :-)

> this.clientConfigurationData.getAuthParams()

Reply:
Got it 👍